pkg/csi/service/vanilla/controller.go

```diff
@@ -553,253 +570,347 @@ func (c *controller) createFileVolume(ctx context.Context, req *csi.CreateVolume
-	err := validateVanillaControllerUnpublishVolumeRequest(ctx, req)
-	if err != nil {
-		msg := fmt.Sprintf("Validation for UnpublishVolume Request: %+v has failed. Error: %v", *req, err)
-		log.Error(msg)
-		return nil, status.Errorf(codes.Internal, msg)
-	}
-	if !strings.Contains(req.VolumeId, ".vmdk") {
-		// check if volume is block or file, skip detach for file volume
-		queryFilter := cnstypes.CnsQueryFilter{
-			VolumeIds: []cnstypes.CnsVolumeId{{Id: req.VolumeId}},
-		}
-		queryResult, err := c.manager.VolumeManager.QueryVolume(ctx, queryFilter)
-		if err != nil {
-			msg := fmt.Sprintf("QueryVolume failed for volumeID: %q. %+v", req.VolumeId, err.Error())
-			log.Error(msg)
-			return nil, status.Error(codes.Internal, msg)
-		}
-		if len(queryResult.Volumes) == 0 {
-			msg := fmt.Sprintf("volumeID %q not found in QueryVolume", req.VolumeId)
-			log.Error(msg)
 			return nil, status.Error(codes.Internal, msg)
 		}
-		if queryResult.Volumes[0].VolumeType == common.FileVolumeType {
-			log.Infof("Skipping ControllerUnpublish for file volume %q", req.VolumeId)
-			return &csi.ControllerUnpublishVolumeResponse{}, nil
-		}
-	} else {
-		// in-tree volume support
-		if !commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIMigration) {
-			// Migration feature switch is disabled
-			msg := fmt.Sprintf("volume-migration feature switch is disabled. Cannot use volume with vmdk path :%q", req.VolumeId)
-			log.Error(msg)
-			return nil, status.Errorf(codes.Internal, msg)
-		}
-		// Migration feature switch is enabled
-		// ControllerUnpublishVolume will never be the first call back for vmdk registration with CNS.
-		// Here in the migration.VolumeSpec, we do not supply SPBM Policy name.
-		// Node drain is the pre-requisite for volume migration, so volume will be registered with SPBM policy
-		// during ControllerPublish if metadata-syncer fails to register volume using associated SPBM Policy.
-		// for ControllerUnpublishVolume we anticipate volume is already registered with CNS, and volumeMigrationService
-		// should return volumeID for requested VolumePath
-		volumePath := req.VolumeId
-		// In case if feature state switch is enabled after controller is deployed, we need to initialize the volumeMigrationService
-		if err := initVolumeMigrationService(ctx, c); err != nil {
-			// Error is already wrapped in CSI error code
-			return nil, err
-		}
-		req.VolumeId, err = volumeMigrationService.GetVolumeID(ctx, &migration.VolumeSpec{VolumePath: volumePath})
+		err = common.DetachVolumeUtil(ctx, c.manager, node, req.VolumeId)
 		if err != nil {
-			msg := fmt.Sprintf("failed to get VolumeID from volumeMigrationService for volumePath: %q", volumePath)
+			msg := fmt.Sprintf("failed to detach disk: %+q from node: %q err %+v", req.VolumeId, req.NodeId, err)
 			log.Error(msg)
-			return nil, status.Errorf(codes.Internal, msg)
+			return nil, status.Error(codes.Internal, msg)
```

Comment:
status.Error(codes.Internal, msg) ==> status.Errorf(codes.Internal, msg). Just for consistency